src/Utilities/vector_bounds.jl

```diff
@@ -0,0 +1,411 @@
+    ci::MOI.ConstraintIndex{MOI.SingleVariable,S},
+) where {T,S}
+    flag = _single_variable_flag(S)
+    b.set_mask[ci.value] &= ~flag
+    if !iszero(flag & _LOWER_BOUND_MASK)
+        b.lower[ci.value] = _no_lower_bound(T)
+    end
+    if !iszero(flag & _UPPER_BOUND_MASK)
+        b.upper[ci.value] = _no_upper_bound(T)
+    end
+    return
+end
+
+function MOI.delete(b::SingleVariableConstraints, x::MOI.VariableIndex)
+    # To "delete" the variable, set it to 0x00 (free).
+    b.set_mask[x.value] = 0x00
+    if b.variable_indices === nothing
+        b.variable_indices = Set(MOI.get(b, MOI.ListOfVariableIndices()))
+    end
+    delete!(b.variable_indices, x)
+    return
+end
+
+function MOI.is_valid(
+    b::SingleVariableConstraints,
+    ci::MOI.ConstraintIndex{MOI.SingleVariable,S},
+) where {S}
+    if !(1 <= ci.value <= length(b.set_mask))
+        return false
+    end
+    return !iszero(b.set_mask[ci.value] & _single_variable_flag(S))
+end
+
+function MOI.set(
+    b::SingleVariableConstraints,
+    ::MOI.ConstraintSet,
+    ci::MOI.ConstraintIndex{MOI.SingleVariable,S},
+    set::S,
+) where {S}
+    flag = _single_variable_flag(S)
+    if !iszero(flag & _LOWER_BOUND_MASK)
+        b.lower[ci.value] = _lower_bound(set)
+    end
+    if !iszero(flag & _UPPER_BOUND_MASK)
+        b.upper[ci.value] = _upper_bound(set)
+    end
+    return
+end
+
+function MOI.get(
+    b::SingleVariableConstraints,
+    ::MOI.NumberOfConstraints{MOI.SingleVariable,S},
+) where {S}
+    flag = _single_variable_flag(S)
+    return count(mask -> !iszero(flag & mask), b.set_mask)
+end
+
+function _add_constraint_type(
+    list,
+    b::SingleVariableConstraints,
+    S::Type{<:MOI.AbstractScalarSet},
+)
+    flag = _single_variable_flag(S)::UInt8
+    if any(mask -> !iszero(flag & mask), b.set_mask)
+        push!(list, (MOI.SingleVariable, S))
+    end
+    return
+end
+
+function MOI.get(
+    b::SingleVariableConstraints{T},
+    ::MOI.ListOfConstraintTypesPresent,
+) where {T}
+    list = Tuple{DataType,DataType}[]
+    _add_constraint_type(list, b, MOI.EqualTo{T})
+    _add_constraint_type(list, b, MOI.GreaterThan{T})
+    _add_constraint_type(list, b, MOI.LessThan{T})
+    _add_constraint_type(list, b, MOI.Interval{T})
+    _add_constraint_type(list, b, MOI.Semicontinuous{T})
+    _add_constraint_type(list, b, MOI.Semiinteger{T})
+    _add_constraint_type(list, b, MOI.Integer)
+    _add_constraint_type(list, b, MOI.ZeroOne)
+    return list
+end
+
+function MOI.get(
+    b::SingleVariableConstraints,
+    ::MOI.ListOfConstraintIndices{MOI.SingleVariable,S},
+) where {S}
+    list = MOI.ConstraintIndex{MOI.SingleVariable,S}[]
+    flag = _single_variable_flag(S)
+    for (index, mask) in enumerate(b.set_mask)
+        if !iszero(mask & flag)
+            push!(list, MOI.ConstraintIndex{MOI.SingleVariable,S}(index))
+        end
+    end
+    return list
+end
+
+###
+### MatrixBounds
+###
+
+"""
+    struct MatrixBounds{T} <: AbstractVectorBounds
```

Comment:
This is confusing, it seems in the name that it is the bound on each entry of a matrix. I find it weird to name something with what is using it.

Reply:
Open to bike shed. Suggestions for a better name?

Reply:
Not really :/

Reply:
Renamed to `Hyperrectangle`